doc/designs/evp_skey.md

```diff
@@ -0,0 +1,183 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represented as octet buffer. It
+may be not relevant for the situation of hardware-backed providers, such as
+PKCS#11 or TPM.
```

Comment:
Add: in these case the key is generally kept secret on the token and cannot be exported.